### ISSUE TITLE
Nested UL tags not working

### ISSUE THREAD

**Issue by MoeHamoud** [PrivacyApps/html-textview#29]
For some reason nested UL tags aren't rendering correctly. When I use a WebView, they render perfectly. Now, I've tried using WebViews but the performance has been terrible and I've even encountered app crashes.

Here's a sample of my HTML. 

```
<ul>
    <li>Blah blah blah blah blah</li>
</ul>
<ul>
    <ul>
        <li>Blah blah blah blah blah</li>
    </ul>
</ul>
<ul>
    <ul>
        <li>Blah blah blah blah blah</li>
    </ul>
</ul>
<ul>
    <ul>
        <li>Blah blah blah blah blah</sup>
        </li>
    </ul>
</ul>
```

And what it should look like:

<ul>
    <li>Blah blah blah blah blah</li>
</ul>

<ul>
    <ul>
        <li>Blah blah blah blah blah</li>
    </ul>
</ul>

<ul>
    <ul>
        <li>Blah blah blah blah blah</li>
    </ul>
</ul>

<ul>
    <ul>
        <li>Blah blah blah blah blah</sup>
        </li>
    </ul>
</ul>

Except I'm getting this:

<ul>
    <li>No Compromise, Fully Capable</li>
        <li>Seating for up to 7 passengers</li>
        <li>Significantly updated exterior and interior</li>
        <li>New Platinum trim level<sup>1</sup>
        </li>
</ul>

Is there anything that I can do to get this working properly with your library ?

EDIT: Fixed formatting.

**Comment by dschuermann** [comment]
Nested tags are not really tested. I am open for pull request to fix this.

**Comment by Kisty** [comment]
Might this commit be relevant? https://github.com/android/platform_frameworks_base/commit/8b36c0bbd1503c61c111feac939193c47f812190

**Comment by dschuermann** [comment]
Yep that is interesting. Means we shouldn't implement our own lists handling for Android > N

**Comment by MoeHamoud** [comment]
Interesting indeed. I'll test later on N and see whether it handles them well, fingers crossed that this is finally it. Thanks @Kisty !

**Comment by Kisty** [comment]
Have come across this nested list bug yesterday and I fixed it by making a HTML file and putting it in res/raw, then used a WebView to load the page.

Out of interest, how would I go about handling a nested list in this library? Would I add functionality to the HtmlTagHandler?

**Comment by MoeHamoud** [comment]
@Kisty You're right. WebViews will handle the nested tags but WebViews themselves have their own set of performance issues. I've experienced loading issues and various different problems on different devices when I used WebViews. 

html-textview came to the rescue. I used this library alongside my own parser (jsoup helped) for nested tags (its really ugly) and it worked out quite well.

As for handling the nested list, yes you would add the functionality to HtmlTagHandler. @dschuermann  is open to pull requests and I've been meaning to tackle this issue but haven't had the time.
